File: backend/app/mock_data.py
"""
Mock data script to populate the database with initial data for development
"""
import datetime
import json
from sqlalchemy.orm import Session

from app.core.database import engine, Base, get_db
from app.models.canteen import Canteen
from app.models.menu_item import MenuItem
from app.models.user import User
from app.models.cart import Cart, CartItem
from app.models.order import Order, OrderItem, OrderStep

def add_mock_users(db: Session):
    """Add mock users to the database"""
    users = [
        User(id="a1b2c3d4-e5f6-7890-abcd-ef1234567890", name="John Doe", email="john@example.com", password="hashedpassword", role="user", favoriteCanteens=[], recentOrders=[]),
        User(id="b2c3d4e5-f6a7-8901-bcde-f12345678901", name="Jane Smith", email="jane@example.com", password="hashedpassword", role="user", favoriteCanteens=[], recentOrders=[]),
    ]
    
    for user in users:
        db_user = db.query(User).filter(User.id == user.id).first()
        if not db_user:
            db.add(user)
    
    db.commit()
    print("✅ Mock users added to database")

def add_mock_canteens(db: Session):
    """Add mock canteens to the database"""
    canteens = [
        Canteen(id=1, name="Central Canteen", location="Main Building", openTime="08:00", closeTime="20:00"),
        Canteen(id=2, name="Library Cafe", location="Library Building", openTime="09:00", closeTime="18:00"),
        Canteen(id=3, name="Tech Hub Canteen", location="Technology Block", openTime="07:30", closeTime="22:00")
    ]
    
    for canteen in canteens:
        db_canteen = db.query(Canteen).filter(Canteen.id == canteen.id).first()
        if not db_canteen:
            db.add(canteen)
    
    db.commit()
    print("✅ Mock canteens added to database")

def add_mock_menu_items(db: Session):
    """Add mock menu items to the database"""
    menu_items = [
        MenuItem(
            id=1, name="Paneer Butter Masala", description="Rich and creamy paneer curry",
            price=180.0, image="https://images.unsplash.com/photo-1567188040759-fb8a254b3bd2?q=80&w=300&auto=format&fit=crop",
            category="Indian Delights", canteenId=1, isAvailable=True, tags=["Vegetarian"], isPopular=True
        ),
        MenuItem(
            id=2, name="Masala Dosa", description="Crispy crepe filled with spicy potato filling",
            price=80.0, image="https://images.unsplash.com/photo-1589301760014-d929f86731c7?q=80&w=300&auto=format&fit=crop",
            category="South Indian", canteenId=1, isAvailable=True, tags=["Vegetarian"], isPopular=False
        ),
        MenuItem(
            id=3, name="Cold Coffee", description="Refreshing cold coffee with ice cream",
            price=70.0, image="https://images.unsplash.com/photo-1594631252845-29fc4cc8cde9?q=80&w=300&auto=format&fit=crop",
            category="Beverages", canteenId=2, isAvailable=True, tags=["Vegetarian"], isPopular=True
        ),
        MenuItem(
            id=4, name="Veg Burger", description="Delicious vegetable patty with fresh veggies",
            price=90.0, image="https://images.unsplash.com/photo-1550317138-10000687a72b?q=80&w=300&auto=format&fit=crop",
            category="Fast Food", canteenId=3, isAvailable=True, tags=["Vegetarian"], isPopular=True
        ),
        MenuItem(
            id=5, name="French Fries", description="Crispy potato fries with seasoning",
            price=60.0, image="https://images.unsplash.com/photo-1541592106381-b31e9677c0e5?q=80&w=300&auto=format&fit=crop",
            category="Fast Food", canteenId=3, isAvailable=True, tags=["Vegetarian"], isPopular=False
        ),
    ]
    
    for item in menu_items:
        db_item = db.query(MenuItem).filter(MenuItem.id == item.id).first()
        if not db_item:
            db.add(item)
    
    db.commit()
    print("✅ Mock menu items added to database")

def add_mock_cart_data(db: Session):
    """Add mock cart data to the database"""
    # Create cart for user 1
    cart1 = db.query(Cart).filter(Cart.userId == "a1b2c3d4-e5f6-7890-abcd-ef1234567890").first()
    if not cart1:
        cart1 = Cart(
            userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
            createdAt=datetime.datetime.utcnow().isoformat(),
            updatedAt=datetime.datetime.utcnow().isoformat()
        )
        db.add(cart1)
        db.flush()  # Get the ID without committing
    
    # Clear any existing cart items for this cart
    db.query(CartItem).filter(CartItem.cartId == cart1.id).delete()
    
    # Add items to cart 1
    cart_items = [
        CartItem(
            cartId=cart1.id,
            menuItemId=1,
            quantity=1,
            selectedExtras=json.dumps({
                "Spice Level": "Medium",
                "Extra Paneer": True
            }),
            specialInstructions="Extra spicy"
        ),
        CartItem(
            cartId=cart1.id,
            menuItemId=3,
            quantity=2,
            selectedExtras=json.dumps({
                "Sugar": "Less",
                "Ice": "Regular"
            }),
            specialInstructions="Extra cold"
        ),
        CartItem(
            cartId=cart1.id,
            menuItemId=4,
            quantity=1,
            selectedExtras=json.dumps({
                "Extra Cheese": True,
                "No Onion": True
            }),
            specialInstructions="Well done"
        )
    ]
    
    for item in cart_items:
        db.add(item)
    
    # Create cart for user 2
    cart2 = db.query(Cart).filter(Cart.userId == "b2c3d4e5-f6a7-8901-bcde-f12345678901").first()
    if not cart2:
        cart2 = Cart(
            userId="b2c3d4e5-f6a7-8901-bcde-f12345678901",
            createdAt=datetime.datetime.utcnow().isoformat(),
            updatedAt=datetime.datetime.utcnow().isoformat()
        )
        db.add(cart2)
        db.flush()  # Get the ID without committing
    
    # Clear any existing cart items for this cart
    db.query(CartItem).filter(CartItem.cartId == cart2.id).delete()
    
    # Add items to cart 2
    cart_items = [
        CartItem(
            cartId=cart2.id,
            menuItemId=2,
            quantity=2,
            selectedExtras=json.dumps({
                "Extra Chutney": True
            }),
            specialInstructions="Extra crispy"
        ),
        CartItem(
            cartId=cart2.id,
            menuItemId=5,
            quantity=1,
            selectedExtras=json.dumps({
                "Extra Salt": True,
                "Sauce": "Ketchup"
            }),
            specialInstructions="Extra crispy"
        )
    ]
    
    for item in cart_items:
        db.add(item)
    
    db.commit()
    print("✅ Mock cart data added to database")

def add_mock_orders(db: Session):
    """Add mock orders to the database"""
    # Clear existing order data
    db.query(OrderStep).delete()
    db.query(OrderItem).delete()
    db.query(Order).delete()
    
    # Active order for current date (April 17, 2025)
    active_order = Order(
        userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
        canteenId=1,
        totalAmount=450.00,
        status="Processing",
        orderTime=datetime.datetime.strptime("2025-04-16T12:15:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        paymentMethod="Card",
        paymentStatus="Paid",
        phone="1234567890",
        pickupTime="12:45 PM",
        isPreOrder=False
    )
    db.add(active_order)
    db.flush()

    # Order steps for active order
    order_steps = [
        OrderStep(
            orderId=active_order.id,
            status="Order Placed",
            description="Your order has been received by the vendor.",
            time="12:15 PM",
            completed=1,
            current=0
        ),
        OrderStep(
            orderId=active_order.id,
            status="Preparing",
            description="The kitchen is preparing your food.",
            time="12:20 PM",
            completed=0,
            current=1
        ),
        OrderStep(
            orderId=active_order.id,
            status="Ready for Pickup",
            description="Your order is ready for pickup.",
            time="",
            completed=0,
            current=0
        ),
        OrderStep(
            orderId=active_order.id,
            status="Completed",
            description="Your order has been picked up.",
            time="",
            completed=0,
            current=0
        )
    ]
    for step in order_steps:
        db.add(step)
        
    # Order items for active order
    order_items = [
        OrderItem(
            orderId=active_order.id,
            itemId=1,
            quantity=1,
            customizations=json.dumps(["Extra Butter", "Medium Spicy"]),
            note="Make it extra creamy"
        ),
        OrderItem(
            orderId=active_order.id,
            itemId=2,
            quantity=2,
            customizations=json.dumps([]),
            note=""
        ),
        OrderItem(
            orderId=active_order.id,
            itemId=3,
            quantity=1,
            customizations=json.dumps(["Large Portion"]),
            note=""
        )
    ]
    for item in order_items:
        db.add(item)
    
    # Order history
    order_history1 = Order(
        userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
        canteenId=1,
        totalAmount=320.00,
        status="Completed",
        orderTime=datetime.datetime.strptime("2025-04-12T15:30:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        paymentMethod="Card",
        paymentStatus="Paid",
        phone="1234567890",
        isPreOrder=False
    )
    db.add(order_history1)
    db.flush()
    
    order_history2 = Order(
        userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
        canteenId=2,
        totalAmount=280.00,
        status="Completed",
        orderTime=datetime.datetime.strptime("2025-04-10T13:15:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        paymentMethod="Card",
        paymentStatus="Paid",
        phone="1234567890",
        isPreOrder=False
    )
    db.add(order_history2)
    db.flush()
    
    order_history3 = Order(
        userId="a1b2c3d4-e5f6-7890-abcd-ef1234567890",
        canteenId=3,
        totalAmount=520.00,
        status="Cancelled",
        orderTime=datetime.datetime.strptime("2025-04-05T19:30:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        cancelledTime=datetime.datetime.strptime("2025-04-05T19:40:00.000Z", "%Y-%m-%dT%H:%M:%S.%fZ").isoformat(),
        cancellationReason="Changed my mind",
        paymentMethod="Cash",
        paymentStatus="Refunded",
        phone="1234567890",
        isPreOrder=False
    )
    db.add(order_history3)
    db.flush()
    
    # Add items for first history order
    history_items1 = [
        OrderItem(
            orderId=order_history1.id,
            itemId=1,
            quantity=1,
            customizations=json.dumps(["Extra Spicy", "Regular Portion"]),
            note=""
        ),
        OrderItem(
            orderId=order_history1.id,
            itemId=2,
            quantity=2,
            customizations=json.dumps([]),
            note=""
        ),
        OrderItem(
            orderId=order_history1.id,
            itemId=3,
            quantity=1,
            customizations=json.dumps([]),
            note=""
        )
    ]
    
    # Add items for second history order
    history_items2 = [
        OrderItem(
            orderId=order_history2.id,
            itemId=4,
            quantity=1,
            customizations=json.dumps(["No Onions", "Extra Cheese"]),
            note=""
        ),
        OrderItem(
            orderId=order_history2.id,
            itemId=5,
            quantity=1,
            customizations=json.dumps(["Extra Salt"]),
            note=""
        ),
        OrderItem(
            orderId=order_history2.id,
            itemId=3,
            quantity=1,
            customizations=json.dumps([]),
            note=""
        )
    ]
    
    # Add items for third history order
    history_items3 = [
        OrderItem(
            orderId=order_history3.id,
            itemId=1,
            quantity=1,
            customizations=json.dumps(["Extra Veggies"]),
            note=""
        ),
        OrderItem(
            orderId=order_history3.id,
            itemId=2,
            quantity=1,
            customizations=json.dumps(["Dry", "Extra Spicy"]),
            note=""
        ),
        OrderItem(
            orderId=order_history3.id,
            itemId=3,
            quantity=1,
            customizations=json.dumps([]),
            note=""
        ),
        OrderItem(
            orderId=order_history3.id,
            itemId=4,
            quantity=1,
            customizations=json.dumps(["No Eggs"]),
            note=""
        )
    ]
    
    # Add all history order items
    for item in history_items1 + history_items2 + history_items3:
        db.add(item)
    
    db.commit()
    print("✅ Mock orders added to database")

def initialize_mock_data():
    """Initialize database with mock data"""
    try:
        # Create tables if they don't exist
        Base.metadata.create_all(bind=engine)
        print("✅ Database tables created")
        
        # Add mock data
        db = next(get_db())
        add_mock_users(db)
        add_mock_canteens(db)
        add_mock_menu_items(db)
        add_mock_cart_data(db)
        add_mock_orders(db)
        
        print("✅ All mock data added successfully")
    except Exception as e:
        print(f"❌ Error initializing mock data: {e}")

if __name__ == "__main__":
    initialize_mock_data()

File: frontend/src/data/mockData.ts
// Mock data for the Smart Canteen application

// Canteens
export const canteens = [
  {
    id: 1,
    name: "Central Canteen",
    image: "https://images.unsplash.com/photo-1517248135467-4c7edcad34c4?q=80&w=2070&auto=format&fit=crop",
    location: "Main Campus Building",
    rating: 4.2,
    openTime: "07:30",
    closeTime: "22:00",
    isOpen: true,
    description: "The main canteen serving a variety of cuisines including Indian, Chinese, and Continental."
  },
  {
    id: 2,
    name: "Tech Hub Cafe",
    image: "https://images.unsplash.com/photo-1554118811-1e0d58224f24?q=80&w=2047&auto=format&fit=crop",
    location: "Technology Block",
    rating: 4.5,
    openTime: "08:00",
    closeTime: "20:00",
    isOpen: true,
    description: "Modern cafe with quick bites, beverages, and light meals for tech students."
  },
  {
    id: 3,
    name: "Hostel Mess",
    image: "https://images.unsplash.com/photo-1590846406792-0adc7f938f1d?q=80&w=1972&auto=format&fit=crop",
    location: "Hostel Complex",
    rating: 3.8,
    openTime: "07:00",
    closeTime: "21:30",
    isOpen: true,
    description: "The main dining facility for hostel residents with breakfast, lunch, and dinner options."
  },
  {
    id: 4,
    name: "Faculty Lounge",
    image: "https://images.unsplash.com/photo-1559925393-8be0ec4767c8?q=80&w=2071&auto=format&fit=crop",
    location: "Admin Block",
    rating: 4.7,
    openTime: "09:00",
    closeTime: "18:00",
    isOpen: false,
    description: "Exclusive dining area for faculty and staff with premium food options."
  }
];

// Food Categories
export const categories = [
  { id: 1, name: "Breakfast", icon: "coffee" },
  { id: 2, name: "Lunch", icon: "utensils" },
  { id: 3, name: "Dinner", icon: "moon" },
  { id: 4, name: "Snacks", icon: "cookie" },
  { id: 5, name: "Beverages", icon: "mug-hot" },
  { id: 6, name: "Desserts", icon: "ice-cream" }
];

// Food Items
export const menuItems = [
  {
    id: "101",
    canteenId: "1",
    canteenName: "Faculty Lounge",
    name: "Masala Dosa",
    description: "Crispy rice crepe filled with spiced potato mixture, served with sambar and chutney",
    price: 60,
    category: "Breakfast",
    image: "https://images.unsplash.com/photo-1589301760014-d929f3979dbc?q=80&w=2070&auto=format&fit=crop",
    tags: ["South Indian", "Vegetarian"],
    rating: 4.5,
    ratingCount: 120,
    isAvailable: true,
    preparationTime: 15,
    isPopular: true,
    customizationOptions: {
      sizes: [
        { name: "small" as const, price: 50 },
        { name: "medium" as const, price: 60 },
        { name: "large" as const, price: 70 },
      ],
      additions: [
        { name: "Extra Chutney", price: 10 },
        { name: "Ghee Roast", price: 15 },
      ],
      removals: ["Onions", "Green Chilies"],
    },
  },
  {
    id: "102",
    canteenId: "2",
    canteenName: "Faculty Lounge",
    name: "Chole Bhature",
    description: "Spicy chickpea curry served with deep-fried bread",
    price: 80,
    category: "Lunch",
    image: "https://images.unsplash.com/photo-1589352911312-5d218efc96be?q=80&w=1974&auto=format&fit=crop",
    tags: ["North Indian", "Vegetarian"],
    rating: 4.3,
    ratingCount: 95,
    isAvailable: true,
    preparationTime: 20,
    isPopular: true,
    customizationOptions: {
      sizes: [
        { name: "small" as const, price: 70 },
        { name: "medium" as const, price: 80 },
        { name: "large" as const, price: 90 },
      ],
      additions: [
        { name: "Extra Bhature", price: 20 },
        { name: "Onions on Side", price: 0 },
      ],
      removals: ["Spices"],
    },
  },
  {
    id: "103",
    canteenId: "1",
    canteenName: "Central Canteen",
    name: "Chicken Biryani",
    description: "Fragrant basmati rice cooked with chicken, spices, and herbs",
    price: 120,
    category: "Lunch",
    image: "https://images.unsplash.com/photo-1589309736404-be8c25f8dea8?q=80&w=1974&auto=format&fit=crop",
    tags: ["Hyderabadi", "Non-Vegetarian"],
    rating: 4.7,
    ratingCount: 150,
    isAvailable: true,
    preparationTime: 30,
    isPopular: true,
    customizationOptions: {
      sizes: [
        { name: "small" as const, price: 100 },
        { name: "medium" as const, price: 120 },
        { name: "large" as const, price: 140 },
      ],
      additions: [
        { name: "Extra Raita", price: 15 },
        { name: "Extra Spicy", price: 0 },
      ],
      removals: ["Coriander"],
    },
  },
  {
    id: "104",
    canteenId: "1",
    canteenName: "Central Canteen",
    name: "Veg Pulao",
    description: "Basmati rice cooked with mixed vegetables and mild spices",
    price: 90,
    category: "Dinner",
    image: "https://images.unsplash.com/photo-1596797038530-2c107aa4606c?q=80&w=1935&auto=format&fit=crop",
    tags: ["North Indian", "Vegetarian"],
    rating: 4.0,
    ratingCount: 80,
    isAvailable: false,
    preparationTime: 25,
    isPopular: false,
    customizationOptions: {
      sizes: [
        { name: "small" as const, price: 80 },
        { name: "medium" as const, price: 90 },
        { name: "large" as const, price: 100 },
      ],
      additions: [{ name: "Extra Raita", price: 15 }],
      removals: ["Peas"],
    },
  },
];

// Orders
export const orders = [
  {
    id: 1001,
    userId: 1,
    canteenId: 1,
    items: [
      { itemId: 101, quantity: 2, customizations: [1] },
      { itemId: 103, quantity: 1, customizations: [5, 6] }
    ],
    totalAmount: 240,
    status: "delivered",
    orderTime: "2023-04-10T08:30:00",
    deliveryTime: "2023-04-10T08:45:00"
  },
  {
    id: 1002,
    userId: 1,
    canteenId: 2,
    items: [
      { itemId: 201, quantity: 1, customizations: [8] },
      { itemId: 202, quantity: 2, customizations: [] }
    ],
    totalAmount: 205,
    status: "preparing",
    orderTime: "2023-04-10T13:15:00",
    deliveryTime: null
  }
];

// User data
export const userData = {
  id: 1,
  name: "Aryan Kumar",
  email: "aryan.kumar@example.edu",
  role: "student",
  favoriteCanteens: [1, 2],
  recentOrders: [1001, 1002]
};

// Order status options
export const orderStatusOptions = [
  "pending", 
  "confirmed", 
  "preparing", 
  "ready", 
  "delivered", 
  "cancelled"
];
